tests/fix/ClassnameUsedAsString/asArgument/output.txt

```diff
@@ -10,9 +10,9 @@ function classname_function(classname<C> $cls): void {}
 function caller(): void {
     $c = new C();
     $int = 5;
-    C::static_string_method($int, nameof C, 5.6);
-    string_function(nameof C);
-    $c->string_method(nameof C, 4);
+    C::static_string_method($int, nameof \C, 5.6);
+    string_function(nameof \C);
```

Comment:
Will this break with `nameof TGeneric`? I think we could do something smart where `\Foo::class` is replaced with `nameof Foo` without having to add this, by adjusting what we replace and using different offsets

Reply:
Good point. https://github.com/facebook/hhvm/blob/master/hphp/hack/manual/hack/03-expressions-and-operators/34-nameof.md says reified generics are not supported as a target for `nameof`. This leads to a funny situation:
```hack
class A {

}

class B<T as A> {
	public function foo(): void {
		// typechecks
		echo T::class;
		// does not typecheck
		echo nameof T;
	}
}

class C<reify T as A> {
	public function foo(): void {
		// typechecks
		echo T::class;
		// does not typecheck
		echo nameof T;
	}
}
```
I think we need to exclude these from the autofix.

Reply:
It seems this is already fine because for `T::class` we receive a `TMixedWithFlags` as the type rather than a `TLiteralClassname`. I will add tests.